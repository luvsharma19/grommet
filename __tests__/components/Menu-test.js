// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import Menu from '../../src/js/components/Menu';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('Menu', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Menu>
        <a href="#" className="active">
          First
        </a>
        <a href="#">
          Second
        </a>
        <a href="#">
          Third
        </a>
      </Menu>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Generate the menu within the DOM of the parent div', () => {
    let divRef;
    const component = renderer.create(
      <div ref={ref=> divRef = ref}>
        <Menu dropContainer={divRef}>
          <a href="#" className="active">
            First
          </a>
          <a href="#">
            Second
          </a>
          <a href="#">
            Third
          </a>
        </Menu>
      </div>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
