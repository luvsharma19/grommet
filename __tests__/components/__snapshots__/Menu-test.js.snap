// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Generate the menu within the DOM of the parent div 1`] = `
<div>
  <nav
    className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-menu grommetux-menu--column grommetux-menu--inline"
    dropContainer={undefined}
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}
  >
    <a
      className="active"
      href="#"
    >
      First
    </a>
    <a
      href="#"
    >
      Second
    </a>
    <a
      href="#"
    >
      Third
    </a>
  </nav>
</div>
`;

exports[`Menu has correct default options 1`] = `
<nav
  className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-menu grommetux-menu--column grommetux-menu--inline"
  dropContainer={undefined}
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}
>
  <a
    className="active"
    href="#"
  >
    First
  </a>
  <a
    href="#"
  >
    Second
  </a>
  <a
    href="#"
  >
    Third
  </a>
</nav>
`;
